### PR TITLE
Ncnews 1

### PR DESCRIPTION
**Pull request by Darren80**
Task 3 and task 4

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+const app = express();
+app.use(express.json());
+
+
+app.get('/api/topics', getTopics);
+app.get('/api/articles', getArticles);
+app.get('/api/article/:id', getArticleById);
+
+listen(app);
```
Comment (MrMcSnail) [inline app.js:13]: No need for listen in your app as supertest attaches its own listener during testing

File: models/models.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+const format = require('pg-format');
+const db = require('../db/connection');
+
+const fetchTopics = () => { 
+    return db.query('SELECT * FROM topics;')
+    .then(topics => {
+        return topics.rows;
+    })
+    .catch(err => {
```
Comment (MrMcSnail) [inline models/models.js:9]: good work moving all those catch blocks

File: models/models.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        console.log(err);
+    });
+}
+
+const fetchArticles = () => {
```
Comment (MrMcSnail) [inline models/models.js:14]: This query should also fetch a `comment_count` which is the total count of all the comments with this `article_id` - you should make use of queries to the database in order to achieve this.

File: app.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const express = require('express');
+const { getTopics, getArticles, getArticleById } = require('./controllers/controllers.js');
+const listen = require('./listen');
+
+const app = express();
+app.use(express.json());
```
Comment (MrMcSnail) [inline app.js:6]: This is not needed for the current functionality.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+describe('GET app.js', () => {
+    test('GET /api/topics', () => {
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then((response) => {
+            response = response.body;
+
+            expect(response.length).toBeGreaterThan(0);
```
Comment (MrMcSnail) [inline __tests__/app.test.js:14]: nice

**Review by MrMcSnail — APPROVED** [review]
Merge Away! 
Please make new branches for each pull  request from now on. 

Next step for improvement is to make your commit messages descriptive of the exact functionality or improvement that that specific commit adds. As if before the commit message it reads 'If accepted, this commit will..... < commit message here >'

File: app.js (hunk 1 above)
Comment (Darren80) [reply app.js:13]: I do like to test my app's output by viewing it using localhost in the browser. No longer works when I remove listen, is that ok?

File: app.js (hunk 4 above)
Comment (MrMcSnail) [reply app.js:6]: Yes.
just FYI, we won't be able to keep track of replies to comments. If you get really stuck or if anything is unclear, send a slack or ncHelp

**Pull request merged**